### DISCUSSION TITLE
wire:loading.attr doesn't behave the same as wire:loading visibility

### DISCUSSION THREAD

**Discussion by stuartcusackie** [livewire/livewire#3073] [7. Bugs (archived)]
### Description

wire:loading.attr="disabled" ONLY seems to be applied when actions are running.
However, wire:loading visibility applies BOTH when properties are being updated and when actions are running.
Am I missing something or does this seem inconsistent? Perhaps, there's a flaw in my code. Can someone confirm this behavior?

### Exact steps to reproduce

1. Set up a component with a property (name) and an action (submit). 
2. Set up a template with an input and button (code below)

### Stripped-down, copy-pastable code snippets

```
// Of course I would defer this input usually. I'm just using it as an example
<input type="text" wire:model="name" />

<button class="bg-black text-white disabled:opacity-25" wire:loading.attr="disabled" type="button" wire:click="submit">
    <span wire:loading>Loading...</span>
    <span wire:loading.remove>Submit</span>
</button>
```

If you set up the code as explained then you will notice:
- On submit: button becomes disabled AND shows 'Loading...' text
- On change of input / property: button ONLY shows 'Loading...' text

**Comment by ryangjchandler** [comment]
I'm noticing the same problem. Haven't checked what's causing it yet.

**Comment by sinuoy** [comment]
The bug still persists
